src/main/java/oilinjection/domain/RentCompany.java

```diff
@@ -0,0 +1,59 @@
+package oilinjection.domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import oilinjection.domain.car.Car;
+import oilinjection.domain.vo.RentInfo;
+
+public class RentCompany {
+
+    private static final String QUANTITY_OVER_EXCEPTION_MESSAGE_FORMAT = "[ERROR] %s의 예약 가능한 차량이 없습니다.";
+    private static final String REPORT_FORMAT = "%s : %.0f리터%n";
+
+    private final List<Car> rents;
+
+    public RentCompany() {
+        this.rents = new ArrayList<>();
+    }
+
+    public void acceptReservation(final List<RentInfo> rentInfos) {
+        validateIsPossibleReservation(rentInfos);
+
+        final List<Car> rentCars = rentInfos.stream()
+            .map(this::rent)
+            .collect(Collectors.toList());
+
+        rents.addAll(Collections.unmodifiableList(rentCars));
+    }
+
+    private Car rent(RentInfo rentInfo) {
+        return RentCar.rent(rentInfo.getType(), rentInfo.getTripDistance());
+    }
+
+    private void validateIsPossibleReservation(final List<RentInfo> rentInfos) {
+        rentInfos.stream()
+            .collect(Collectors.groupingBy(RentInfo::getType, Collectors.summingInt(x -> 1)))
```

Comment:
stream중에는 `Collectors.counting()`이라는 게 있습니다! 이것을 활용해볼 수 있을 것 같아요~
```suggestion
            .collect(Collectors.groupingBy(RentInfo::getType, Collectors.counting()))
```

Reply:
아하! `Collectors.counting()`을 보니 Long 타입으로 변환되기에 `validatePerCarType의 매개변수`를 `Long으로 변경`하고 내부 로직에서 `형변환`하는 형식으로 변경하였습니다!